index.js

```diff
@@ -7,35 +7,55 @@ var request = require('request-promise');
 * var driver = new Driver('https://username:key@localhost:3000');
 */
 
+var returnBody = res => {
+  return res.data;
+};
 
-Driver = function(AUTH_URL) {
+var Driver = function(AUTH_URL) {
+  var creds = AUTH_URL.split('@')[0].split('//')[1];
```

Comment:
Perhaps check for `AUTH_URL` to be supplied in the correct format so this won't trip up if it's not.